concent_api/core/payments/report.py

```diff
@@ -0,0 +1,759 @@
+            errors,
+            warnings,
+            matched_subtask_result_with_additional_verification_payment_for_requestor,
+            additional_verification_payment,
+        )
+
+    for additional_verification_payment in payments.additional_verification_payments_for_provider:
+        matched_subtask_result_with_additional_verification_payment_for_provider = find_subtask_result_for_additional_verification_payment(
+            additional_verification_payment,
+            responses.subtask_results_settled,
+            responses.subtask_results_rejected,
+        )
+
+        check_additional_verification_payment_errors_and_warning(
+            errors,
+            warnings,
+            matched_subtask_result_with_additional_verification_payment_for_provider,
+            additional_verification_payment,
+        )
+
+    for force_payment_committed_pending_response in responses.force_payment_committed:
+        matched_settlement_payment_for_force_payment_committed = find_settlement_payment_for_force_payment_committed(
+            force_payment_committed_pending_response,
+            payments.settlement_payments,
+        )
+
+        error_was_added = check_error_cannot_match_message_with_any_payment(
+            errors,
+            matched_settlement_payment_for_force_payment_committed,
+            force_payment_committed_pending_response,
+        )
+
+        if error_was_added:
+            check_warning_deposit_claim_exists_for_missing_match(
+                warnings,
+                force_payment_committed_pending_response.subtask.subtask_id,
+                ConcentUseCase.FORCED_PAYMENT,
+            )
+
+        check_error_more_matching_payment_for_message_than_given_amount(
+            errors,
+            matched_settlement_payment_for_force_payment_committed,
+            force_payment_committed_pending_response,
+        )
+        check_error_message_value_is_different_than_payment_value(
+            errors,
+            matched_settlement_payment_for_force_payment_committed,
+            force_payment_committed_pending_response,
+        )
+        check_error_message_payment_ts_is_different_than_payment_closure_time(
+            errors,
+            matched_settlement_payment_for_force_payment_committed,
+            force_payment_committed_pending_response,
+        )
+        check_warning_force_payment_committed_indicates_that_deposit_was_too_low_to_cover_whole_cost(
+            warnings,
+            force_payment_committed_pending_response,
+        )
+
+    for forced_subtask_payment in payments.forced_subtask_payments:
+        check_error_deposit_claim_exists_for_forced_payment(
+            errors,
+            forced_subtask_payment.tx_hash,
+        )
+
+    for settlement_payment in payments.settlement_payments:
+        check_error_deposit_claim_exists_for_settlement_payment(
+            errors,
+            settlement_payment.tx_hash,
+        )
+
+    return errors, warnings
```

Comment:
I would change into `NamedTuple`

Reply:
Answered in https://github.com/golemfactory/concent/pull/1073#discussion_r250418559